src/modules/callouts/pages/CreateCallout.vue

```diff
@@ -0,0 +1,145 @@
+<template>
+  <h4>Create a new callout</h4>
+  <div class="grid grid-cols-4 gap-8">
+    <div class="col-span-1">
+      <Stepper :steps="steps" v-model:selectedStepIndex="selectedStepIndex" />
+    </div>
+    <div class="col-span-3">
+      <component
+        :is="selectedStep.component"
+        v-model:data="selectedStep.data"
+        v-model:validated="selectedStep.validated"
+      ></component>
+
+      <div class="flex mt-5">
+        <AppButton
+          variant="linkOutlined"
+          :disabled="selectedStep === steps[0]"
+          @click="selectedStepIndex--"
+          >Back</AppButton
+        >
+        <AppButton
+          class="ml-2"
+          v-show="selectedStepIndex != steps.length - 1"
+          :disabled="!selectedStep.validated"
+          @click="selectedStepIndex++"
+          >Continue</AppButton
+        >
+        <AppButton
+          class="ml-2"
+          v-show="selectedStepIndex === steps.length - 1"
+          :disabled="!selectedStep.validated"
+          @click="submitForm"
+          >Submit</AppButton
+        >
+      </div>
+    </div>
+  </div>
+</template>
+
+<script lang="ts" setup>
+import { reactive, ref, computed, markRaw } from 'vue';
+import Stepper from '../components/Stepper.vue';
+import Visibility from '../components/Visibility.vue';
+import TitleAndImage from '../components/TitleAndImage.vue';
+import EndMessage from '../components/EndMessage.vue';
+import AppButton from '../../../components/forms/AppButton.vue';
+import UrlAndSharing from '../components/UrlAndSharing.vue';
+import MailchimpSync from '../components/MailchimpSync.vue';
+import DatesAndDurationVue from '../components/DatesAndDuration.vue';
+import ContentStep from '../components/ContentStep.vue';
+
+export type Step = {
+  id: string;
+  name: string;
+  description: string;
+  validated: boolean;
+  component: any;
+  data: {};
+};
+export type Steps = Array<Step>;
+
+const steps: Steps = reactive([
+  {
+    id: 'content',
+    name: 'Callout content',
+    description: 'Build your callout with questions and prompts',
+    validated: false,
+    component: markRaw(ContentStep),
```

Comment:
Presumably because `ContentStep` itself will be updating/changing while the user interacts with it?

Reply:
This just fixes the error in the console about marking a component as reactive. `markRaw` just tells Vue to exclude the component. It's really only the `data` and `validated` properties that need to be reactive as the rest don't change in any case.

> Vue received a Component which was made a reactive object. This can lead to unnecessary performance overhead, and should be avoided by marking the component with markRaw or using shallowRef instead of ref.